estuary/ui
# Prevent field selection auto test during edit to prevent save logs from showing

## Issues

Fixes https://github.com/estuary/ui/issues/1051

## Changes

### 1051

- Detect if we are going from `Saving` to `Tested` and skip it.
- Added some eventing to check for this in LR

## Tests

### Manually tested

- I explained how I tested the defect in the issue
- Also tested normal create/edit to make sure happy path was still good

### Automated tests

- N/A

## Screenshots

We have a `Skipped` event where we skip updating the form because that update is no longer needed. This is coming in because we have already started `saving` while the `testing_background` is running/done.
**note** : I updated `Skipped` to use `Prevented` and pass a `type` to make LR searching easier
![image](https://github.com/estuary/ui/assets/270078/9638169a-08bb-4e4f-ba31-dc74d2115bbe)

### Regression Tests
Capture Create
![image](https://github.com/estuary/ui/assets/270078/7e7019c0-5067-4394-a2ea-aad211b020be)

Capture Edit
![image](https://github.com/estuary/ui/assets/270078/4644451c-95f3-471c-80f9-daab7ab84078)

Field Selection Refresh
![image](https://github.com/estuary/ui/assets/270078/b3f30e86-8ee7-40c4-a9b3-b4919c66c8d8)
